torch/testing/_internal/common_methods_invocations.py

```diff
@@ -546,6 +547,28 @@ def sample_inputs_pinverse(op_info, device, dtype, requires_grad=False):
     return out
 
 
+def sample_inputs_flip(op_info, device, dtype, requires_grad):
+    test_cases = (
+        make_tensor((S, M, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
+        make_tensor((S, 0, M), device, dtype, low=None, high=None, requires_grad=requires_grad)
+    )
+
+    dims = [[0, 1, 2], [0], [0, 2], [-1]]
+    out = []
+    for dim in dims:
+        for case in test_cases:
+            random.shuffle(dim)
```

Comment:
Let's make a few tweaks here:

- rename `test_cases` to `tensors`
- add back the case where dim is `()`
- make `dims` a tuple of tuples
- rename `out` to `samples`
- iterate over a product of `tensors` and `dims`
- remove the randomization by killing the `random.shuffle`
- I think with the above changes you won't have to `tuple(dim)`

Reply:
Interestingly,

```python
>>> a = torch.randn(3,2,0,2)
>>> torch.flip(a, dims=())
tensor([], size=(3, 2, 0, 2))
>>> b = a.cuda()
>>> torch.flip(a, dims=())
tensor([], size=(3, 2, 0, 2))
>>> torch.flip(b, dims=())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: flip dims size out of range, got flip dims size=0
```

Should I just add a test in `test_shape_ops` which verifies this behavior i.e. works on CPU but assertRaises on CUDA?

Reply:
Why doesn't it work on CUDA?

Reply:
That is, there's probably a bug here:

https://github.com/pytorch/pytorch/blob/b824fc4de22c0e278ec57669b1b82c2d8e1f7394/aten/src/ATen/native/cuda/TensorTransformations.cu#L71

Filing an issue, testing for the current behavior + citing the issue, and fixing this in a future PR would be cool.

Reply:
Sure. Thanks for the reference!